editor/gui/editor_scene_tabs.cpp

```diff
@@ -142,38 +142,44 @@ void EditorSceneTabs::_update_context_menu() {
 	scene_tabs_context_menu->reset_size();
 
 	int tab_id = scene_tabs->get_hovered_tab();
+	bool no_root_node = !EditorNode::get_editor_data().get_edited_scene_root(tab_id);
 
 	scene_tabs_context_menu->add_shortcut(ED_GET_SHORTCUT("editor/new_scene"), EditorNode::FILE_NEW_SCENE);
 	if (tab_id >= 0) {
 		scene_tabs_context_menu->add_shortcut(ED_GET_SHORTCUT("editor/save_scene"), EditorNode::FILE_SAVE_SCENE);
+		_disable_menu_option_if(EditorNode::FILE_SAVE_SCENE, no_root_node);
 		scene_tabs_context_menu->add_shortcut(ED_GET_SHORTCUT("editor/save_scene_as"), EditorNode::FILE_SAVE_AS_SCENE);
+		_disable_menu_option_if(EditorNode::FILE_SAVE_AS_SCENE, no_root_node);
 	}
 	scene_tabs_context_menu->add_shortcut(ED_GET_SHORTCUT("editor/save_all_scenes"), EditorNode::FILE_SAVE_ALL_SCENES);
+	_disable_menu_option_if(EditorNode::FILE_SAVE_ALL_SCENES, EditorNode::get_editor_data().get_edited_scene_count() == 0 || !EditorNode::get_editor_data().get_edited_scene_root(0) || EditorNode::get_editor_data().get_scene_path(0).is_empty());
```

Comment:
Why are we checking only the first scene in the second and the third condition? Shouldn't we iterate through all of them and see if any of them is savable?

Reply:
There can be only 1 scene that isn't savable, it happens when you close all the scenes. Otherwise every scene has either path or root node.

Reply:
I suppose there can be only one, but it doesn't have to be the first one? You can open a new tab without a root node and it can be in any position, can it not?

<img width="232" alt="godot windows editor dev x86_64_2023-09-01_18-40-47" src="https://github.com/godotengine/godot/assets/11782833/550958d6-9ecc-45ff-b99d-01b5bec69b0f">
<img width="278" alt="godot windows editor dev x86_64_2023-09-01_18-41-01" src="https://github.com/godotengine/godot/assets/11782833/b0fb7ec2-8135-4901-bd48-428a65f7b1dd">

Reply:
But then if you save all, you will be prompted to select a file. Since the scene has a root node, it can be saved. You can't have more than 1 completely empty scene.

Reply:
In my screenshots, that's with your PR applied. Depending on the order the option is either disabled or enabled. The empty tab is completely empty and has no root node. Saving when the option is enabled doesn't prompt anything, nothing at all happens.

At the very least it's inconsistent right now.

Reply:
Ah right, now I noticed what your screenshot is about. Tab count check should be `<= 1`.

Reply:
Curiously, the progress is only shown when I try to save all by clicking on the valid tab. It saves with any of the tabs selected, but it doesn't show the progress if the valid tab (in need of saving) is in the background. That's unrelated to this PR, though.